### PR TITLE
Calculator: [Testing] Add Front end tests

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Adds frontend unit tests to the calculator.

## People to notify

@moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
@moollaza This doesn't seem to have triggered Travis for some reason. I'm guessing we need to add `npm test` into https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/.travis.yml#L27 ?

**Comment by pjhampton** [comment]
@mintsoft I used `[skip ci]` in the commit message

**Comment by mintsoft** [comment]
@pjhampton it'll also need a minimal `packages.js` like:
```
{"scripts": {"test": "jasmine-node spec"}}
```
It might need name and version as a minimum too, I'm not too sure

File: .travis.yml [hunk 1]
```diff
@@ -25,4 +25,4 @@ perl:
   - 5.16
 script:
   - prove -lr -j1 t
-  - npm test
+  - jasmine
```
Comment (mintsoft) [inline .travis.yml]: `jasmine-node` IIRC

Comment (pjhampton) [inline .travis.yml]: Sure, I will give it a go

**Comment by pjhampton** [comment]
@mintsoft ugh, screw it. This is on hold anyway because a refactor is needed to test at a high level.

We can phase this in with the Conversions UI. Zaahir had a working branch for inspiration: https://github.com/duckduckgo/zeroclickinfo-goodies/compare/zaahir/jasmine-testing-example

I'm not around for the comm leader call this weekend, but we can have a separate call if you want to pitch in and help standardize front-end tests in DDH?

**Comment by mintsoft** [comment]
@pjhampton I've branched this branch into https://github.com/duckduckgo/zeroclickinfo-goodies/tree/mintsoft/calc-add-tests-fix ( https://github.com/duckduckgo/zeroclickinfo-goodies/compare/pjhampton/calc-add_tests...mintsoft/calc-add-tests-fix?expand=1 ) and the tests are running fine: https://travis-ci.org/duckduckgo/zeroclickinfo-goodies/builds/226119836

**Comment by pjhampton** [comment]
@mintsoft Thanks! 💯  ✨

**Comment by moollaza** [comment]
@mintsoft @pjhampton we shouldn't need `node-jasmine` or to use `npm test`, checkout my branch where I first tested this out:

https://github.com/duckduckgo/zeroclickinfo-goodies/compare/zaahir/jasmine-testing-example

You can see it runs OK at the end of the build: 
https://travis-ci.org/duckduckgo/zeroclickinfo-goodies/builds/225971026#L1009

The jasmine docs suggest you just `npm install jasmine` and then execute `jasmine` -- Perhaps the `npm` approach is older?

**Comment by moollaza** [comment]
@pjhampton @mintsoft looks like it's working now, but the test is failing 👍

**Comment by mintsoft** [comment]
@moollaza definitely could be an older approach, it's one I've used previously with the same results as you've got: https://travis-ci.org/duckduckgo/zeroclickinfo-goodies/builds/226119836

Either way looks good!

**Comment by pjhampton** [comment]
Killing this off because of

  - https://github.com/duckduckgo/p5-app-duckpan/pull/388
  - https://github.com/duckduckgo/zeroclickinfo-goodies/pull/4160

Will reimplement FE tests for the calculator in a separate PR

**Pull request closed without merging**